kratos/python_scripts/read_materials_process.py

```diff
@@ -111,40 +108,32 @@ def _AssignPropertyBlock(self, data):
 
         # Set the CONSTITUTIVE_LAW for the current properties.
         if "Variables" in mat["constitutive_law"].keys(): #pass the list of variables when constructing the constitutive law
-           constitutive_law = self._GetItemFromModule( mat["constitutive_law"]["name"])(mat["constitutive_law"]["Variables"])
+           constitutive_law = self._GetItemFromModule( mat["constitutive_law"]["name"].GetString())(mat["constitutive_law"]["Variables"])
         else:
-           constitutive_law = self._GetItemFromModule( mat["constitutive_law"]["name"])()
+           constitutive_law = self._GetItemFromModule( mat["constitutive_law"]["name"].GetString())()
            
         prop.SetValue(CONSTITUTIVE_LAW, constitutive_law)
         
         # Add / override the values of material parameters in the properties
         for key, value in mat["Variables"].items():
             var = self._GetItemFromModule(key)
-            if isinstance(value, (list, tuple)):
-                size_1 = len(value)
-                if isinstance(value[0], (list, tuple)):
-                    size_2 = len(value[0])
-                    matrix = Matrix(size_1,size_2)
-                    for i in range(size_1):
-                        for j in range(size_2):
-                            matrix[i, j] = value[i][j]
-                    prop.SetValue( var, matrix)
-                else:
-                    vector = Vector(size_1)
-                    for i in range(size_1):
-                        vector[i] = value[i]
-                    prop.SetValue( var, vector)
+            if value.IsMatrix():
+                prop.SetValue( var, value.GetMatrix() )
+            elif value.IsVector():
+                prop.SetValue( var, value.GetVector() )
             else:
```

Comment:
I think it is missing the cases when value is of type String and Boolean

Reply:
Good point!
I changed it accordingly, now you can use all variable types that are available in Kratos::Parameters